js/graphql.md

```diff
@@ -0,0 +1,1346 @@
+        id
+        title
+        comments {
+          items {
+            id
+            content
+          }
+        }
+      }
+    }
+  }
+}
+
+# List all blogs, their posts, and their posts comments.
+query ListBlogs {
+  listBlogs { # Try adding: listBlog(filter: { name: { eq: "My New Blog!" } })
+    items {
+      id
+      name
+      posts { # or try adding: posts(filter: { title: { eq: "My Post!" } })
+        items {
+          id
+          title
+          comments { # and so on ...
+            items {
+              id
+              content
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+If you want to update your API, open your project's `backend/api/~apiname~/schema.graphql` file (NOT the one in the `backend/api/~apiname~/build` folder) and edit it in your favorite code editor. You can compile the `backend/api/~apiname~/schema.graphql` by running:
+
+```
+amplify api gql-compile
+```
+
+and view the compiled schema output in `backend/api/~apiname~/build/schema.graphql`.
+
+You can then push updated changes with:
+
+```
+amplify push
+```
+
+## Directives
+
+### @model
+
+Object types that are annotated with `@model` are top-level entities in the
+generated API. Objects annotated with `@model` are stored in DynamoDB and are
+capable of being protected via `@auth`, related to other objects via `@connection`,
+and streamed into Elasticsearch via `@searchable`.
+
+#### Definition
+
+```
+directive @model(
+    queries: ModelQueryMap, 
+    mutations: ModelMutationMap
+) on OBJECT
+input ModelMutationMap { create: String, update: String, delete: String }
+input ModelQueryMap { get: String, list: String }
+```
+
+#### Usage
+
+Define a GraphQL object type and annotate it with the `@model` directive to store
+objects of that type in DynamoDB and automatically configure CRUDL queries and
+mutations.
+
+```
+type Post @model {
+    id: ID! # id: ID! is a required attribute.
+    title: String!
+    tags: [String!]!
+}
+```
+
+You may also override the names of any generated queries and mutations, or remove operations entirely.
+
+```
+type Post @model(queries: { get: "post" }, mutations: null) {
+    id: ID!
+    title: String!
+    tags: [String!]!
+}
+```
+
+This would create and configure a single query field `post(id: ID!): Post` and
+no mutation fields.
+
+#### Generates
+
+A single `@model` directive configures the following AWS resources:
+
+- An Amazon DynamoDB table with 5 r/w units. Support for auto-scaling and encryption at rest coming soon.
+- An AWS AppSync DataSource configured to access the table above.
+- An AWS IAM role attached to the DataSource that allows AWS AppSync to call the above table on your behalf.
+- Up to 8 resolvers (create, update, delete, get, list, onCreate, onUpdate, onDelete) but this is configurable via the `query`, `mutation`, and `subscription` arguments on the `@model` directive.
+
+This input schema document
+
+```
+type Post @model {
+    id: ID!
+    title: String
```

Comment:
Did you mean to have `metadata: MetaData` as a field here on the Post type? Otherwise, the transformation wouldn't put it there, I don't think.